Common/SAM.h

```diff
@@ -339,8 +339,15 @@ struct SAMRecord : SAMAlignment {
 			>> o.cigar >> o.mrnm >> o.mpos >> o.isize;
 #if SAM_SEQ_QUAL
 		in >> o.seq >> o.qual;
+		std::string allTags = "";
 		if (in.peek() != '\n')
-			in >> o.tags;
+			in >> allTags;
+		while (in.peek() != '\n' && !in.eof()) {
+		    std::string tag = "";
+			in >> tag;
+			allTags += "\t" + tag;
+		}
+		o.tags = allTags;
```

Comment:
I believe that you can replace this entire chunk with
```c++
getline(in, o.tags);
```

There are some style red flags here. `!in.eof()` is big ⚠️. Usually you should instead use either `in` in a bool context (like `while (in)`) or more explicitly `!in.fail()`. The former bool context is more idiomatic.

Allocating a `std::string` on the heap in an an inner loop on line 346 is a performance / efficiency no no.
String concatenation which uses the heap in an inner loop on line 348 is a performance / efficiency no no.

Reply:
Yeah I figured, it wasn't the best implementation which is I waited for your input before I merged. Can you explain your comment about why string concatenation in the inner loop is bad?

Reply:
You want to avoid unnecessary heap allocations and frees in an inner loop when possible. The operation `allTags +="\t" + tag` allocates a new string, copies `"\t"` to it and then resizes that string and copies `tag` to it, resizes `allTags`, and copies the temporary string to it, then frees the temporary string, then frees `tag`. That's a lot of unnecessary work and access to the heap in an inner loop. Accesses to the heap in multithreaded code may involve locks and contention.